community/kernel/src/main/java/org/neo4j/kernel/builtinprocs/AwaitIndexProcedure.java

```diff
@@ -139,4 +142,13 @@ private String formatIndex( String labelName, String propertyKeyName )
     {
         return format( ":%s(%s)", labelName, propertyKeyName );
     }
+
+    @Override
+    public void close()
+    {
+        if ( statement != null )
+        {
+            statement.close();
```

Comment:
Honestly do not remember why if is there - removed it completely and add simple test that we close statement on procedure close